algorithms/preprocessing.py

```diff
@@ -1,7 +1,30 @@
 from supporting_components.graph import Graph, Vertex
+from algorithms.color_initialization import degree_color_initialization
+from algorithms.decide_gi import __is_balanced
```

Comment:
Het lijkt net alsof de __is_balanced een private functie is van deze file, maar dat is niet zo.

Reply:
Zoals hij nu geschreven is is het wel een private functie, maar je kunt hem wel aanroepen. Ik zou in decide_gi wel de naam van deze methode kunnen aanpassen zodat je ook alleen kan opvragen of iets balanced is. Dat een idee?

Reply:
Of kan ik beter gewoon de is_balanced_or_bijected methode gebruiken? Want als ik is_balanced gebruikt, geeft hij naast de boolean ook een lijst met colornums terug die we gebruiken in is_bijected. Dat is het enige lelijke aan wat ik nu doe

Maar als ik de is_balanced_or_bijected gebruikt, doet hij meer berekeningen dan nodig zijn.